app/src/features/interact/components/InteractionDrawer.tsx

```diff
@@ -15,11 +17,15 @@ function InteractionDrawer({
   contractId,
   updateLog,
 }: InteractionDrawerProps) {
+
+  // Import theme state
+  const theme = useThemeContext().theme;
   return (
     <Drawer
       PaperProps={{
         sx: {
-          background: '#F1F1F1',
+          color: theme === 'light' ? darkColors.gray1 : lightColors.scalesGreen7,
```

Comment:
Rather than having to check `theme === 'light' ?` everywhere, a cleaner solution might be to create a hashmap of the colors used in the app with the mappings for the mode (in theme.ts). For example:

```typescript
interface ColorMapping {
    light: string;
    dark: string;
}

type ColorName = 'gray1' | 'gray2'; // ...names for other colors

const COLORS: Record<ColorName, ColorMapping> = {
    'gray1': {
          'light': darkColors.gray1,
           'dark': lightColors.scalesGreen7,
     }
     // ... all colors used in the app
}

export const useThemeContext = () => {
  {
    const themeContext = useContext(ThemeContext);

    if (!themeContext) {
      throw new Error('Should be wrapped with ThemeContext.Provider!');
    }

    // Convenience function for getting a color based on the current theme
    const themeColor = useCallback((name: ColorName) => COLORS[name][theme], [theme]);

    return {
      setTheme: themeContext.setTheme,
      theme: themeContext.theme,
      themeColor,
    };
  }
};
```

Then in this file you can do: 

```tsx
  const { themeColor } = useThemeContext();
  return (
    <Drawer
      PaperProps={{
        sx: {
          color: themeColor('gray1'),
```

This will also make it easier to ensure that the styles between light and dark mode are "mirrored" everywhere.

Reply:
Thank you for suggesting it, this is really an extremely convenient approach, done that.